src/lib.rs

```diff
@@ -906,13 +908,31 @@ impl Pandoc {
         // apply all filters
         let filtered = filters.into_iter().fold(o, |acc, item| item(acc));
         self.input = Some(InputKind::Pipe(filtered));
+        Ok(())
+    }
+    
+    /// actually execute pandoc
+    pub fn execute(mut self) -> Result<(), PandocError> {
+        let _ = try!(self.preprocess());
         let _ = try!(self.run());
         Ok(())
     }
+
+    /// Execute pandoc and return the result from stdout.
+    pub fn execute_with_output(mut self) -> Result<String, PandocError> {
+        let _ = try!(self.preprocess());
+        let output = try!(self.run());
+        match String::from_utf8(output) {
+            Ok(string) => Ok(string),
+            Err(err) => Err(PandocError::from(err.utf8_error())),
+        }
+    }
 }
 
 /// Possible errors that can occur before or during pandoc execution
 pub enum PandocError {
+    /// conversion from UTF-8 failed; includes valid-up-to byte count.
+    BadUtf8Conversion(usize),
```

Comment:
I should note that one other alternative is simply to do `String::from_utf8_lossy(output)` [above](https://github.com/oli-obk/rust-pandoc/pull/7/files#diff-b4aea3e418ccdb71239b96952d9cddb6R925), where invalid UTF-8 will just end up as � in the resulting output. That may be the best of the options.